compiler/src/dotty/tools/dotc/typer/ErrorReporting.scala

```diff
@@ -28,11 +28,14 @@ object ErrorReporting {
 
   def cyclicErrorMsg(ex: CyclicReference)(implicit ctx: Context) = {
     val cycleSym = ex.denot.symbol
-    def errorMsg(msg: String, cx: Context): Message =
+    def errorMsg(msg: Message, cx: Context): Message =
       if (cx.mode is Mode.InferringReturnType) {
         cx.tree match {
           case tree: untpd.DefDef if !tree.tpt.typeOpt.exists =>
-            OverloadedOrRecursiveMethodNeedsResultType(tree.name)
+            // TODO: analysis if tree is an overloaded method (or directly recursive)
```

Comment:
According to that stacktrace the error is _not_ caused by the call 

    cycleSym.owner.info.member(cycleSym.name).isOverloaded

That one shows up nowhere. So it most likely happens later. Could it be that an error message is not reported?